Initialize backend directory

nadmax opened the pull request:
## Description
This PR sets up the initial backend for Terrascope, a stateless microservice that parses Terraform state files and generates interactive infrastructure graphs.

## What's new?
**Backend API**
- Parser: Converts `terraform.tfstate` into a normalized graph structure with nodes (resources) and edges (dependencies)
- HTTP Endpoints:
  - GET `/health` - Health check with version info, uptime, and system details
  - POST `/parse` - Accepts tfstate and returns graph (nodes + edges), with a conditional pretty print
- Models: Type-safe structs for Terraform state and graph representation
- Docker: Multi-stage build with Alpine runtime for minimal image size (~15MB)

**CI/CD**
- GitHub Actions workflow with:
  - Docker image build (tagged with commit SHA)
  - Go tests with module caching (go.mod version detection)
  - Sequential execution: build → test

**Infrastructure**
- compose.yml for local development
- Test files co-located with implementation (*_test.go)

nadmax committed:
feat(backend): init backend directory

sofianedjerbi commented on backend/cmd/api/main_test.go:
C'est pas vital mais si jamais le fichier devient trop gros tu peux le découper en sous-fichiers (je vois qu'il test un peu plus que ce qu'il y a dans main

C'est quoi la convention en go ?

sofianedjerbi commented on backend/internal/parser/graph.go:
```diff
@@ -0,0 +1,116 @@
+				graph.Edges = append(graph.Edges, edge)
+			}
+
+			for _, dep := range instance.Dependencies {
```
Entre res.DependsOn et instance.Dependencies, tu risques les doublons ?

sofianedjerbi commented on compose.yml:
```diff
@@ -0,0 +1,22 @@
+      test:
+        [
+          "CMD",
+          "wget",
```
team curl.......

nadmax replied:
> Oui je sais, mais sur les alpine il n'y a pas curl d'installé... C'est pour ça que j'a fait avec wget.
> 
> J'ajouterai les changements au prochain commit

Bon finalement pas de curl, parce que c'est pas opti sur la taille de l'image

sofianedjerbi reviewed (changes requested):
Good, j'ai survolé les tests parcontre (j'ai juste lu les titres, je fais confiance à toi et à la CI sur la logique..)

nadmax commented on backend/internal/models/graph.go:
```diff
@@ -0,0 +1,31 @@
+}
+
+type Node struct {
+	ID       string         `json:"id"`
```
L'ID correspond au nom de la ressource, par exemple `aws_vpc`

nadmax commented on backend/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+FROM golang:1.25.3-alpine AS builder
+WORKDIR /app
+COPY go.mod ./
+RUN go mod download
```
Oui mais en réalité c'est une erreur de ma part...
Il faut copier le go.sum pour s'assurer d'avoir les bonnes versions des dépendances.

Ça sera dans mon prochain commit :)

sofianedjerbi commented on backend/go.mod:
```diff
@@ -0,0 +1,11 @@
+require github.com/stretchr/testify v1.11.1
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
```
C'est pas automatique ? :fearful:

nadmax committed:
fix(parser): remove extra dot before index in buildNodeID function

Previously, node IDs were generated with an extra dot before the instance index,
resulting in identifiers like `aws_subnet.private.[0]` instead of the expected
`aws_subnet.private[0]`.

The function now concatenates the index directly to the resource name, matching
Terraform’s native addressing format and ensuring consistent node identifiers.

sofianedjerbi reviewed (approved):
GOOD !
